bvm/ballerina-core/src/main/java/org/ballerinalang/persistence/serializable/serializer/providers/instance/SerializableBMapInstanceProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.persistence.serializable.serializer.providers.instance;
+
+import org.ballerinalang.model.util.serializer.TypeInstanceProvider;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.persistence.serializable.SerializableState;
+import org.ballerinalang.persistence.serializable.reftypes.impl.SerializableBMap;
+
+/**
+ * Provide object instance to represent {@link SerializableBMap}.
+ *
+ * @since 0.98.1
+ */
+public class SerializableBMapInstanceProvider implements TypeInstanceProvider {
+    @Override
+    public String getTypeName() {
+        return SerializableBMap.class.getSimpleName();
+    }
+
+    @Override
+    public Object newInstance() {
+        return new SerializableBMap<>(new BMap<>(), new SerializableState(null, 0));
```

Comment:
Wouldn't be better to create an overloaded constructor for `SerializableState` which only accept the ip?

Then we can replace all the places we pass `null` (`new SerializableState(null, 0)`)

Reply:
Purpose of newInstance() method is to return an instance of the type under consideration, without altering that type's code. All the fields of this instance will be overwritten later in the deserialization process.

Reply:
My point is, if we are passing null to a constructor, that is an issue with the constructor itself. Because that constructor would be written assuming there will be a value for that argument, and may cause issues if we pass null.

Ideally there should be a constructor which does not accept that argument.